lib/collection/header.js

```diff
@@ -190,6 +192,38 @@ _.extend(Header, /** @lends Header */ {
         var args = Array.prototype.slice.call(arguments);
         args.unshift(Header);
         return new (Header.bind.apply(Header, args))();
+    },
+
+    /**
+     * Gets value from an Array or {@link PropertyList} of Headers by filtering through key.
+     *
+     * @param {Array|PropertyList<Header>} headers
+     * @param {string} key
+     * @returns {string}
+     */
+    getHeaderValue: function(headers, key) {
+        if (!_.isArray(headers) && !PropertyList.isPropertyList(headers)) {
+            throw new Error('It is only possible to unparse PropertyList or Array of headers.');
+        }
+        var header = _.find(headers, function (header) {
+            return header.key === key;
+        });
+        return header ? header.value : null;
+    },
+
+    /**
+     * Gets size of an Array or {@link PropertyList} of Headers.
+     *
+     * @param {Array|PropertyList<Header>} headers
+     * @returns {Number}
+     */
+    getSize: function(headers) {
+        if (!_.isArray(headers) && !PropertyList.isPropertyList(headers)) {
+            throw new Error('It is only possible to unparse PropertyList or Array of headers.');
```

Comment:
No throws please :) 

Check out some of the other classes (`Url`) for what to return when an incorrect type is supplied.